WordPress/gutenberg
# Add confirm dialog before Permanently delete

## What?
Fixes https://github.com/WordPress/gutenberg/issues/60383, where the "Permanently Delete" action for trashed posts in the WordPress editor was not functioning as expected. This PR improves the logic and introduces a confirmation modal to ensure users can confirm their actions before permanently deleting posts.

## Why?
The current implementation of the "Permanently Delete" action did not provide a clear user experience and lacked safeguards like a confirmation modal. This could lead to unintentional deletions. By addressing this issue, the PR ensures a better and more accessible user experience while reducing the likelihood of errors.

## How?
Added a modal confirmation step to confirm the deletion of selected posts.

## Testing Instructions

- Open the WordPress admin dashboard.
- Navigate to any list of posts.
- Move one or more posts to the trash.
- Select the trashed posts and use the "Permanently Delete" option.
- Confirm the deletion in the modal dialog.
- Observe the success or error notices for both individual and bulk deletion scenarios.


## Screenshots or screencast 


https://github.com/user-attachments/assets/09a09a0c-0d0f-417a-ac78-c69db68febf4
